src/Peptides.cpp

```diff
@@ -16,230 +16,364 @@
+          scrambledPeptide = mutatedPeptide;
+        } while ( peptideUsed && ++tries < maxTries_);
+      }
+      //if (tries == maxTries_) cerr << "Gave up on peptide " << it->first << endl;
+    }
+    if(isVerbose_ && scrambledPeptide.length() > minLen_){
+        logger << it->first << ',' << scrambledPeptide << ',' << scrambledPeptide.length() << ',' << wasMutated << '\n';
     }
-    if (tries==maxTries) cerr << "Gave up on peptide " << it->first << endl;
-    pep2ixs[scram].insert(it->second.begin(),it->second.end());
+    pep2ixs_[scrambledPeptide].insert(it->second.begin(),it->second.end());
   }
 }
 
 int Peptides::run() {
-  srand(time(0));
-  readFasta(inFile);
-  Peptides fake;
-  fake.shuffle(*this);
-  fake.printAll();
+  srand(seed_);
+  rGen_.seed(seed_);
+
+  std::ofstream outFileStream;
+  if (!outFile_.empty()) {
+    outFileStream.open(outFile_.c_str(), ios::out);
+  }
+  std::ostream &outStream = !outFile_.empty() ? outFileStream : std::cout;
+
+  std::ofstream logger("shuffle.log", ios::trunc);
+  logger << "original" << ',' << "scrambled" << ',' << "length" << ',' << "wasMutated" << '\n';
+
+  cerr << "Reading fasta file and in-silico digesting proteins" << endl;
+  readFasta(inFile_, prependOriginal_, outStream);
+  background_.print(cerr);
+  for (unsigned int m = 0; m < multFactor_; ++m) {
+    Peptides entrapmentDB(minLen_, usedPeptides_, background_, rGen_, maxTries_, replaceI_);
+
+    cerr << "Shuffling round: " << (m+1) << endl;
+    entrapmentDB.shuffle(pep2ixs_, logger);
+
+    ostringstream suffix("");
+    if (multFactor_ > 1) {
+      suffix << "|shuffle_" << (m+1);
+    }
+
+    entrapmentDB.printAll(connectorStrings_, suffix.str(), outStream);
+    usedPeptides_ = entrapmentDB.usedPeptides_;
+      rGen_ = entrapmentDB.rGen_;
+  }
   return 0;
 }
 
 bool Peptides::parseOptions(int argc, char **argv){
   ostringstream intro;
   intro << "Usage:" << endl;
-  intro << "   mimic <fasta-file> [<fasta-file 2> [<fasta-file 3>[...]]]" << endl;
+  intro << "   mimic <fasta-file>" << endl;
   CommandLineParser cmd(intro.str());
+
+  cmd.defineOption("o",
+      "out-file",
+      "File to write the mimicked protein sequences to (Default: prints to stdout)",
+      "string");
+  cmd.defineOption("p",
+      "prefix",
+      "Prefix to mimic proteins (Default: \"mimic|Random_\")",
+      "string");
+  cmd.defineOption("m",
+      "mult-factor",
+      "Number of times the database should be multiplied (Default: 1)",
+      "int");
+  cmd.defineOption("s",
+      "shared-pept-ratio",
+      "Ratio of shared peptides that will stay preserved in the mimic database (Default: 0.0)",
+      "int");
+  cmd.defineOption("S",
+      "seed",
+      "Set seed of the random number generator. Default = 1",
+      "int");
+  cmd.defineOption("P",
+      "prepend",
+      "Prepend the original fasta file to the output",
+      "",
+      TRUE_IF_SET);
+    cmd.defineOption("I",
+                     "replaceI",
```

Comment:
When this option is not replaced by the proposed switch then `-` should be used as separator.
```suggestion
                     "replace-I",
```